src/Tutor.Web/Controllers/Administrators/Stakeholders/LearnerController.cs

```diff
@@ -85,4 +102,35 @@ public ActionResult<PagedResult<CourseDto>> GetEnrolledCourses(int id, [FromQuer
         var result = _enrollmentService.GetEnrolledCourses(id, page, pageSize);
         return CreateResponse<Course, CourseDto>(result);
     }
+
+    private Result<Learner> MapLearner(StakeholderAccountDto stakeholderAccount)
+    {
+        Learner learner;
+        try
+        {
+            learner = _mapper.Map<Learner>(stakeholderAccount);
+        }
+        catch (AutoMapperMappingException e)
+        {
+            Error rootError = new Error(e.Message).CausedBy(e);
+            return Result.Fail("Unsupported learner type.").WithError(rootError);
+        }
+        return Result.Ok(learner);
+    }
+
+    private Result<List<Learner>> MapLearners(List<StakeholderAccountDto> stakeholderAccounts)
+    {
+        if (stakeholderAccounts.Select(s => s.LearnerType).Distinct().Count() > 1)
```

Comment:
This whole logic seems like it belongs in the service and should be moved there.

Reply:
Logic is related to DTO mappings and handles mapping exceptions, not sure how to migrate to service level. Any suggestions?

Reply:
How do we handle mapping exceptions in general?

Reply:
I don't think we do. This exception seems to happen only when trying to map to enum value if enum value is invalid. AutoMapper handles invalid type conversion for other values as BadRequests.

Reply:
Hmm, my guess is that this is not the only place that deals with enums at the endpoint level. Was this a fix for the 500 error found by @lukaDoric ?

If this code is necessary we can close the PR and open the one for dev into master that we talked about.